runtime/syntax/qml.vim

```diff
@@ -0,0 +1,1145 @@
+syntax keyword qmlObjectLiteralType WebEngineNewViewRequest
+syntax keyword qmlObjectLiteralType WebEngineNotification
+syntax keyword qmlObjectLiteralType WebEngineProfile
+syntax keyword qmlObjectLiteralType WebEngineScript
+syntax keyword qmlObjectLiteralType WebEngineSettings
+syntax keyword qmlObjectLiteralType WebEngineView
+syntax keyword qmlObjectLiteralType WebSocket
+syntax keyword qmlObjectLiteralType WebSocketServer
+syntax keyword qmlObjectLiteralType WebView
+syntax keyword qmlObjectLiteralType WebViewLoadRequest
+syntax keyword qmlObjectLiteralType WeekNumberColumn
+syntax keyword qmlObjectLiteralType WheelEvent
+syntax keyword qmlObjectLiteralType WheelHandler
+syntax keyword qmlObjectLiteralType Window
+syntax keyword qmlObjectLiteralType WlScaler
+syntax keyword qmlObjectLiteralType WlShell
+syntax keyword qmlObjectLiteralType WlShellSurface
+syntax keyword qmlObjectLiteralType WorkerScript
+
+syntax keyword qmlObjectLiteralType XAnimator
+syntax keyword qmlObjectLiteralType XdgDecorationManagerV1
+syntax keyword qmlObjectLiteralType XdgOutputManagerV1
+syntax keyword qmlObjectLiteralType XdgPopup
+syntax keyword qmlObjectLiteralType XdgPopupV5
+syntax keyword qmlObjectLiteralType XdgPopupV6
+syntax keyword qmlObjectLiteralType XdgShell
+syntax keyword qmlObjectLiteralType XdgShellV5
+syntax keyword qmlObjectLiteralType XdgShellV6
+syntax keyword qmlObjectLiteralType XdgSurface
+syntax keyword qmlObjectLiteralType XdgSurfaceV5
+syntax keyword qmlObjectLiteralType XdgSurfaceV6
+syntax keyword qmlObjectLiteralType XdgToplevel
+syntax keyword qmlObjectLiteralType XdgToplevelV6
+syntax keyword qmlObjectLiteralType XmlListModel
+syntax keyword qmlObjectLiteralType XmlRole
+syntax keyword qmlObjectLiteralType XYPoint
+syntax keyword qmlObjectLiteralType XYSeries
+
+syntax keyword qmlObjectLiteralType YAnimator
+
+syntax keyword qmlObjectLiteralType ZoomBlur
+
+" }}}
+
+if get(g:, 'qml_fold', 0)
+  syn match   qmlFunction      "\<function\>"
+  syn region  qmlFunctionFold  start="\<function\>.*[^};]$" end="^\z1}.*$" transparent fold keepend
+
+  syn sync match qmlSync  grouphere qmlFunctionFold "\<function\>"
+  syn sync match qmlSync  grouphere NONE "^}"
+
+  setlocal foldmethod=syntax
+  setlocal foldtext=getline(v:foldstart)
+else
+  syn keyword qmlFunction         function
+  syn match   qmlArrowFunction    "=>"
+  syn match   qmlBraces           "[{}\[\]]"
+  syn match   qmlParens           "[()]"
+endif
+
+syn sync fromstart
+syn sync maxlines=100
+
+if main_syntax == "qml"
+  syn sync ccomment qmlComment
+endif
+
+" Define the default highlighting.
+" For version 5.7 and earlier: only when not done already
+" For version 5.8 and later: only when an item doesn't have highlighting yet
+if version >= 508 || !exists("did_qml_syn_inits")
+  if version < 508
+    let did_qml_syn_inits = 1
+    command -nargs=+ HiLink hi link <args>
+  else
+    command -nargs=+ HiLink hi def link <args>
+  endif
```

Comment:
The HiLink command has been removed from the distributed syntax files as Vim 6 was released over twenty years ago.

These lines can be reduced to, for example, `hi def link qmlComment Comment`

See https://github.com/vim/vim/blob/master/runtime/syntax/aap.vim for an example.